src/Raven.Client/Documents/Session/AsyncDocumentQuery.cs

```diff
@@ -1107,11 +1107,19 @@ internal AsyncDocumentQuery<TResult> CreateDocumentQueryInternal<TResult>(QueryD
                 newFieldsToFetch = FieldsToFetchToken.Create(fields, queryData.Projections.ToArray(), queryData.IsCustomFunction, sourceAlias);
             }
             else
+            {
                 newFieldsToFetch = null;
+                if(FieldsToFetchToken != null)
```

Comment:
do we have this issue with `DocumentQuery` as well?

Reply:
After a quick check, yes, we have!
 I've applied the same fix on `DocumentQuery `and some lines to to test it too.